structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/mapper/EnvironmentRequestProcessingStepMapper.java

```diff
@@ -25,6 +25,6 @@ public boolean isFirstStep(FlowDetails flow) {
     }
 
     public boolean isLastStep(FlowDetails flow) {
-        return "FINAL_STATE".equals(flow.getNextFlowState());
+        return flow.getNextFlowState() != null && (flow.getNextFlowState().endsWith("_FAILED_STATE") || flow.getNextFlowState().endsWith("_FINISHED_STATE"));
```

Comment:
This is going to miss a few flows, in some cases the flows are not ending with FAILED_STATE or FINISHED_STATE, but some cases they are ending with FINALIZE etc.

Reply:
Oh, this is Environment, so we might be good here.